nautobot/dcim/templates/dcim/device.html

```diff
@@ -148,27 +148,30 @@
                                         <td>Primary IPv4</td>
                                         <td>
                                             {% if object.primary_ip4 %}
-                                                <a href="{% url 'ipam:ipaddress' pk=object.primary_ip4.pk %}">{{ object.primary_ip4.address.ip }}</a>
+                                                <a href="{% url 'ipam:ipaddress' pk=object.primary_ip4.pk %}" id="ipv4">{{ object.primary_ip4.address.ip }}</a>
                                                 {% if object.primary_ip4.nat_inside %}
                                                     <span>(NAT for {{ object.primary_ip4.nat_inside.address.ip }})</span>
                                                 {% elif object.primary_ip4.nat_outside %}
                                                     <span>(NAT: {{ object.primary_ip4.nat_outside.address.ip }})</span>
                                                 {% endif %}
+                                                <button class="btn" data-clipboard-target="#ipv4"><span class="ui-icon ui-icon-copy"></span></button>
```

Comment:
I think `ui-icon ui-icon-copy` and `ui-icon-clipboard` are jquery-ui icons? It'd probably be more consistent with the rest of our UI to use a Material Design icon, perhaps `mdi mdi-clipboard-outline` or `mdi mdi-content-copy`?

Also, for Bootstrap, `btn` should be paired with appropriate styling class(es). I'd suggest `btn btn-default` or `btn btn-info`. Possibly also add `btn-sm` or `btn-xs` to reduce the button size per @jathanism's comment.